tests/filesystem_tests/test_hdfs_handler.py

```diff
@@ -53,6 +54,40 @@ def test_klist_not_exist(self):
 
         os.environ['PATH'] = path
 
+    def create_dummy_keytab(self, dummy_username):
+        dummy_password = "123"
+        keytab_path = "user.keytab"
+        command = "(echo 'addent -password -p {}@{} -k 1 -e rc4-hmac' &&\
+                  sleep 1 && echo {} && echo write_kt {}) \
+                  | ktutil".format(dummy_username, "dummy_realm",
+                                   dummy_password, keytab_path)
+        pipe = subprocess.Popen(command, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, shell=True)
+        out, err = pipe.communicate()
+        return keytab_path
+
+    def test_get_principel_name_from_keytab(self):
+        dummy_username = "IAmADummy"
+        keytab_path = self.create_dummy_keytab(dummy_username)
+
+        # save the original KRB5_KTNAME
+        if "KRB5_KTNAME" in os.environ:
+            original_krb5_ktname = os.environ['KRB5_KTNAME']
+        else:
+            original_krb5_ktname = None
+
+        os.environ['KRB5_KTNAME'] = keytab_path
+        with chainerio.create_handler(self.fs) as handler:
+            self.assertEqual(dummy_username, handler.username)
+
+        # put KRB5_KTNAME back
+        if original_krb5_ktname is None:
+            del os.environ['KRB5_KTNAME']
+        else:
+            os.environ['KRB5_KTNAME'] = original_krb5_ktname
+
+        chainerio.remove(keytab_path)
```

Comment:
Use `tempfile.TemporaryDirectory()` with it's cleanup capability on `with` statement in case of unexpected test failure.